internal/rancher/client_test.go

```diff
@@ -78,6 +78,34 @@ var _ = Describe("Client", func() {
 				Expect(server.ReceivedRequests()).To(HaveLen(1))
 			})
 		})
+		
+		When("the response is not 201 with response", func() {
```

Comment:
"the token (or provider) is not found"?

Reply:
In this test, 404 is being returned by the provider URL. This could be the URL is not found/invalid or something else. The provider was found since it has a provider URL to call. This test and the next are meant to cover the different logging paths, and make sure the code does not break when arcade is logging the error.